tests/python/contrib/test_verilator/test_mobilenet.py

```diff
@@ -0,0 +1,240 @@
+    model_url = "https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz"
+    model_path = download_testdata(
+        model_url, "mobilenet_v1_1.0_224_quant.tgz", module=["tf", "official"]
+    )
+    model_dir = os.path.dirname(model_path)
+    extract(model_path)
+    tflite_model_file = os.path.join(model_dir, "mobilenet_v1_1.0_224_quant.tflite")
+    tflite_model_buf = open(tflite_model_file, "rb").read()
+    try:
+        import tflite
+
+        return tflite.Model.GetRootAsModel(tflite_model_buf, 0)
+    except AttributeError:
+        import tflite.Model
+
+        return tflite.Model.Model.GetRootAsModel(tflite_model_buf, 0)
+
+
+def get_input_tensor_name():
+    """Return input name."""
+    return "input"
+
+
+def compile_model_to_relay(model):
+    """Compile model to relay.
+
+    Paramters
+    ---------
+    model : Model
+        The input model.
+
+    Returns
+    -------
+    mod: Module
+        The relay module.
+
+    params: Parameters
+        The model parameters.
+    """
+    input_tensor = get_input_tensor_name()
+    input_shape = (1, 224, 224, 3)
+    input_dtype = "uint8"
+    mod, params = relay.frontend.from_tflite(
+        model,
+        shape_dict={input_tensor: input_shape},
+        dtype_dict={input_tensor: input_dtype},
+    )
+    return mod, params
+
+
+def run_model(mod, params=None, opts=None):
+    """Run model.
+
+    Paramters
+    ---------
+    mod: Module
+        The relay module.
+
+    params: Parameters
+        The model parameters.
+
+    opts: Dict
+        The compiler options.
+
+    Returns
+    -------
+    out: Data
+        The output data.
+    """
+    with transform.PassContext(opt_level=3, config={"relay.ext.verilator.options": opts}):
+        lib = relay.build(mod, target="llvm", params=params)
+    module = runtime.GraphModule(lib["default"](tvm.cpu()))
+    image_data = get_real_image(224, 224)
+    input_tensor = get_input_tensor_name()
+    module.set_input(input_tensor, image_data)
+    module.run()
+    out = module.get_output(0).asnumpy()
+    return out
+
+
+def get_labels():
+    """Return labels."""
+    label_file_url = "".join(
+        [
+            "https://raw.githubusercontent.com/",
+            "tensorflow/tensorflow/master/tensorflow/lite/java/demo/",
+            "app/src/main/assets/",
+            "labels_mobilenet_quant_v1_224.txt",
+        ]
+    )
+    label_file = "labels_mobilenet_quant_v1_224.txt"
+    label_path = download_testdata(label_file_url, label_file, module="data")
+    # List of 1001 classes
+    with open(label_path) as f:
+        labels = f.readlines()
+    return labels
+
+
+def check_result(res):
+    """Check prediction against labels."""
+    labels = get_labels()
+    predictions = np.squeeze(res)
+    prediction = np.argmax(predictions)
+    # 387 is the elephant
+    tvm.testing.assert_allclose(prediction, 387, rtol=1e-5, atol=1e-5)
```

Comment:
In this case can't we just assert that prediction is equal to 387?

Reply:
Fixed